test/tools/CodeCoverageAutomation/Start-CodeCoverageRun.ps1

```diff
@@ -196,14 +196,60 @@ try
         $openCoverParams.Add('SuppressQuiet', $true)
     }
 
+    # grab the commitID, we need this to grab the right sources
+    $gitCommitId = & "$psBinPath\powershell.exe" -noprofile -command { $PSVersiontable.GitCommitId }
+    $commitId = $gitCommitId.substring($gitCommitId.LastIndexOf('-g') + 2)
+
+    # download the src directory
+    try
+    {
+        $gitexe = "C:\Program Files\git\bin\git.exe"
+        # operations relative to where the test location is.
+        # some tests rely on source files being available in $outputBaseFolder/test
+        Push-Location $outputBaseFolder
+        # clean up partial repo clone before starting
+        if ( Test-Path "$outputBaseFolder/.git" )
+        {
+            remove-item -force -recurse "${outputBaseFolder}/.git"
+        }
+        if ( Test-Path "$outputBaseFolder/src" )
+        {
+            remove-item -force -recurse "${outputBaseFolder}/src"
+        }
+        Write-LogPassThru -Message "initializing repo in $outputBaseFolder"
+        & $gitexe init
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
+
+        Write-LogPassThru -Message "adding remote"
+        & $gitexe remote add origin https://github.com/PowerShell/PowerShell
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
+
+        Write-LogPassThru -Message "setting sparse-checkout"
+        & $gitexe config core.sparsecheckout true
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
+
+        Write-LogPassThru -Message "pulling sparse repo"
+        "src" | out-file -encoding ascii .git\info\sparse-checkout
+        & $gitexe pull origin master
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
+
+        Write-LogPassThru -Message "checkout commit $commitId"
+        & $gitexe checkout $commitId
+        Write-LogPassThru -Message "git operation return $LASTEXITCODE"
```

Comment:
similar to above

Reply:
yah - cut and paste. I've updated the log messages